ccf_transaction_processor/transaction_processor/pdo_tp.cpp

```diff
@@ -137,14 +163,139 @@ namespace ccfapp
+                    }
+
+                    string ias_signature = enclave_proof_data.signature;
+                    string verification_report_string = enclave_proof_data.verification_report;
+
+                    // Ensure that expected measurements exist in the ledger
+                    auto expected_measurements_view = ctx.tx.rw(enclave_expected_measurements);
+                    auto expected_measurements_r = expected_measurements_view->get(PDO_ENCLAVE_EXPECTED);
+                    if (!expected_measurements_r.has_value()){
+                        return ccf::make_error(
+                            HTTP_STATUS_BAD_REQUEST, ccf::errors::InvalidInput, "Cannot register Enclave in HW mode without registering expected measurements");
+                    }
+
+                    // expected measurements exist, get the value now
+                    auto expected_measurements = expected_measurements_r.value();
+
+                    // verify ias report signature
+                    if (!verify_ias_signature(ias_signature, expected_measurements.ias_public_key, verification_report_string)){
+                        return ccf::make_error(
+                            HTTP_STATUS_BAD_REQUEST, ccf::errors::InvalidInput, "unable to verify IAS report signature for registering enclave");
+                    }
+
+                    // Parse ias verification report JSON string
+                    try {
+                        auto j = nlohmann::json::parse(verification_report_string);
+                        verification_report = j.get<VerificationReport>();
+                    }
+                    catch(...){
+                        return ccf::make_error(
+                            HTTP_STATUS_BAD_REQUEST, ccf::errors::InvalidInput, "Unable to parse IAS verification report");
+                    }
+
+                    string enclave_quote_body_b64 = verification_report.isvEnclaveQuoteBody;
+                    vector<uint8_t> enclave_quote_body_raw = raw_from_b64(enclave_quote_body_b64);
+
+                    // Verify that the verification report EPID pseudonym matches the enclave_persistent_id
+                    if (in.enclave_persistent_id != verification_report.epidPseudonym) {
+                        return ccf::make_error(
+                            HTTP_STATUS_BAD_REQUEST, ccf::errors::InvalidInput, "Enclave attestation report verification Failed. Invalid epid pseudonym");
+                    }
+
+                    // Verify the verification report enclave quote status
+                    transform(verification_report.isvEnclaveQuoteStatus.begin(), verification_report.isvEnclaveQuoteStatus.end(),
+                        verification_report.isvEnclaveQuoteStatus.begin(), ::toupper);
+                    if ((verification_report.isvEnclaveQuoteStatus != OK_QUOTE_STATUS) && (verification_report.isvEnclaveQuoteStatus != GROUP_OUT_OF_DATE_QUOTE_STATUS)) {
```

Comment:
Is there a possibility here to trigger a warning, or maybe just add a comment that group-out-of-date is allowed?
The group-out-of-date status does not indicate a compromise, yet something is off. 
In theory, the consortium should decide what the policy should be here.

Reply:
I guess what we need to do is two things 1. Extend the "valid enclave" policy definition to include aspects about "group-out-of-date" or any other such things, and more importantly, 2. Have a TP rpc, that permits pdo clients to query the "valid policy". As of now, "policy"only includes MREnclave, and everything else is hard-coded. and 2. does not even exist (even in Sawtooth). IMHO, we should take this up in a separate PR.  (Note that the policy can be updated by the CCF consortium per consortium rules).